python-clock solution

cmdellinger opened the pull request:
solution to exercism.io clock exercise in python

cc @barberj

Commit:
clock solution

barberj commented on python/clock/clock.py:
```diff
@@ -0,0 +1,76 @@
+class Clock:
+
+    # helper functions
+    def two_dig_str(self, x):
```
do you want your helper methods to be public?

barberj commented on python/clock/clock.py:
```diff
@@ -0,0 +1,76 @@
+    
+    # init and call behavior
+    def __init__(self, hours, minutes):
+        self.time = self.clock_time(hours, minutes)
```
did you consider initializing and using a tuple for state? Would lend itself easier to manipulation in `add` and `subtract`

barberj commented on python/clock/clock.py:
```diff
@@ -0,0 +1,76 @@
+    # methods
+    def add(self, minutes):
+        self.time = self.clock_time(int(self.time[0:2]), int(self.time[3:]) + minutes)
+        return self.time
```
the test shows passing the results of `add` and `subtract` to `str`. Would you expect the result of `add` to be a `str`?

Commit:
Misc Refactors

Changes:
- changed time from string to list, since tuples are immutable in Python
- eliminated storage of original time
- added __doc__ string

cmdellinger commented:
I refactored the clock class.

cc @barberj

barberj commented on python/clock/clock.py:
```diff
@@ -0,0 +1,90 @@
+        self.time = self.__clock_time(self.time)
+        return self.__clock_str(self.time)
+    
+    def subtract(self, minutes): # -> String (XX:XX)
```
isn't subtract really just adding a negative number?

barberj commented on python/clock/clock.py:
```diff
@@ -0,0 +1,90 @@
+        """ subtract minutes from time and return new 24 hour clock string in form XX:XX """
+        self.time[1] -= minutes
+        self.time = self.__clock_time(self.time)
+        return self.__clock_str(self.time)
```
Why are you returning a str representation in add and subtract?

Commit:
Refactor - stripped code down to pass tests only

stripped code down to barebones of what needs to be there to pass the
tests

cmdellinger commented:
@barberj 
I updated based on your comments. The methods now return a Clock object and subtract just calls add.

barberj commented:
👍
